### DISCUSSION TITLE
HTTPS behind another reverse proxy

### DISCUSSION THREAD

**Discussion by thremilien** [cdubz/pfnj#54] [Q&A]
Hello, I have deployed your app in a proxmox container and set it up behind a reverse proxy (nginx proxy manager) with SSL connection but when I go on the link, the CSS and the JS doesn't load... if I go directly on the ip adress this works perfectly. There is a screenshot of the chromium error logs in attached files. Can you help me ?
![unknown](https://user-images.githubusercontent.com/51836065/153269972-953b5f9e-e538-483c-8758-81b41b76ebe8.png)

**Comment by cdubz** [comment]
Hm yeah that kind of makes sense... the `asset` calls for those assets see HTTP so they create HTTP links instead of HTTPS because the app doesn't know about the proxy.

I haven't used this behind a proxy myself. Does anything in this article help? https://cylab.be/blog/122/using-https-over-a-reverse-proxy-in-laravel

**Comment by thremilien** [comment]
Hello, this isn't working :(
I understood the problem but I don't find any solutions. I have tried the link you send me and another similar but anyone doesn't work. Let me know if you a solution ! I'm available if you need more informations about my setup

**Comment by danielslyman** [comment]
same issue here

**Comment by T012m3n7oR** [comment]
I am also having this occur to my instance.

**Reply by T012m3n7oR** [reply]
This Stack Overflow post sets an environmental variable that may fix this?
https://stackoverflow.com/a/68287406

I do get the JS file to load on the login page, so at the least it's a step in the right direction.

Dev Console still shows the following, but it's a warning not an error.
![image](https://github.com/kcal-app/kcal/assets/48076977/551e211d-d173-4854-bd6e-dc1cd496b25c)

**Reply by T012m3n7oR** [reply]
This can be fixed by added some sort of forced https in the App/providers/AppServiceProvider.php file of the app container.
this is what I used:
```
<?php

namespace App\Providers;

use App\Search\Ingredient;
use CloudCreativity\LaravelJsonApi\LaravelJsonApi;
use Illuminate\Support\ServiceProvider;
use Illuminate\Support\Facades\URL;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Register any application services.
     *
     * @return void
     */
    public function register()
    {
        //
    }

    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        if (strpos(env('APP_URL', ''), 'https') !== false) {
            URL::forceScheme('https');
        }
        LaravelJsonApi::defaultApi('v1');

        if (config('scout.driver') === 'algolia') {
            Ingredient::bootSearchable();
        }
    }
}
```

**Reply by andbez** [reply]
I had the same problem: But

 URL::forceScheme('https'); 
must be
->  \URL::forceScheme('https');

**Comment by Amolith** [comment]
I've had this same issue across many Laravel apps, and the fix I've used in the past has worked here as well. You need to configure your webserver to tell PHP-FPM that the connection is HTTPS, and if I understand correctly, that overrides the URL generation stuff that Laravel uses and forces _all_ URLs to be HTTPS.

I don't know how that would work with NGINX or Apache, but this is the entirety of my Caddy config.
```caddy
:8080 {
        encode zstd gzip
        root * /var/www/kcal/public/
        php_fastcgi unix//run/php/php8.3-fpm.sock {
                env HTTPS true
        }
        file_server
}
```

And I just reverse proxy port 8080.

**Comment by marvin2309** [comment]
Please see [pull request](https://github.com/kcal-app/kcal/pull/69) I convertert @T012m3n7oR into a PR. @cdubz
